trail/query/views.py

```diff
@@ -61,3 +66,226 @@ def print_results(request):
+        return {
+                "wcs_center_x__gte": upperLeft["x"],
+                "wcs_center_x__lte": lowerRight["x"],
+                "wcs_center_y__lte": upperLeft["y"],
+                "wcs_center_y__gte": lowerRight["y"],
+                "wcs_center_z__lte": upperLeft["z"],
+                "wcs_center_z__gte": lowerRight["z"]
+                }
+
+    def isWcsQueryParamMissing(self, queryParams):
+        """
+        Check if the sky boundary is correctly specified
+
+        Parameters
+        ----------
+        queryParams: Dictionary
+            A dictionary that contains the boundary for the sky
+
+        Return
+        ------
+        Return true if any of the param needed is not there
+        """
+        return not (self.RALOW in queryParams and self.RAHIGH in queryParams
+                    and self.DECHIGH in queryParams and self.DECLOW in queryParams)
+
+
+class MetadataQuery(APIView):
+
+    """
+    Support Query that returns metadata entry as a result through metadata info
+
+    Attributes
+    ----------
+    metadataDao: MetadataDao
+        Data accessing objects for metadata table
+
+    """
+
+    DATA_FIELDS = "fields"
+
+    def __init__(self):
+        self.metadataDao = MetadataDAO()
+
+    # gets all the metadata entry that matches the query parameters.
+    @swagger_auto_schema(responses={200: MetadataSerializer(many=True)})
+    def get(self, request):
+
+        """
+        Returns the Metadata entries that meets the users input in json
+
+        Parameters
+        ----------
+        fields: List of strings that is the key of wanted string
+            list of wanted field in the metadata returned
+        queryParam: Dictionary
+            list of criterias for metadatas specified by users
+
+        Return
+        ------
+        A json response that contains the list of metadatas wanted
+
+        """
+        fields = request.query_params.getlist(self.DATA_FIELDS)
+
+        queryParams = request.query_params.dict()
+        queryParams.pop(self.DATA_FIELDS, None)
+        results = self.metadataDao.queryByParams(queryParams)
+        if len(fields) == 0:
+            fields = None
+
+        result = MetadataSerializer(results, many=True, fields=fields)
+
+        return Response(result.data, status=status.HTTP_200_OK)
+
+
+class WcsQuery(APIView):
+    """
+    Support Query that returns metadata entry that is in the part of sky specified by the users.
+
+    Attributes
+    ----------
+    metadataDao: MetadataDao
+        Data accessing objects for metadata table
+
+    """
+
+    def __init__(self):
+        self.metadataDao = MetadataDAO()
+
+    def get(self, request):
+
+        """
+        Returns the Metadata associated with the part of sky the user specifies
+
+        Parameters
+        ----------------
+        raLow: the lower bound for ra
+        raHigh: the upper bound for ra
+        decLow: the lower bound for dec
+        decHigh: the upper bound for dec
```

Comment:
Parameters are things the function actually takes in. In this case:

````
Parameters
---------------
request: `rest_framework.request.Request`
    Request object containing the query parameters which must contain:
    `raLow`, `raHigh`, `decLow` and `decHigh` keys; the values of which 
    set the sky bounding box limits.
```

Also missing a `Returns` section

```
Returns
-----------
metadata : `rest_framework.response.Response`
    JSON response containing `Metadata` entries within the specified
   bounding box.
```